### ISSUE TITLE
Storing function pointer in a void*

### ISSUE THREAD

**Issue by Jiwan** [SuperV1234/Experiments#3]
https://github.com/SuperV1234/Experiments/blob/ec15da53c7cee087e6b72e8023e46e352beb4270/function_ref.cpp#L24

I am not quite sure if you are allowed to store function pointer into ptr_. I believe that you would need an union of some kind for function object (void*) and function pointers ReturnType (*)(ArgTypes...);.

**Comment by SuperV1234** [comment]
You are correct - a proper implementation would require a `union`. This (toy) implementation is currently relying on undefined behavior

I think you would find the "Open Question" sections at the top of the last revision of the paper interesting:
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0792r2.html
